pyomo/core/kernel/expr_coopr3.py

```diff
@@ -638,7 +638,11 @@ def to_string(self, ostream=None, verbose=None, precedence=0):
             ostream.write("( ")
         first = True
         if self._coef != 1:
+            if self._coef < 0:
+                ostream.write("(")
             ostream.write(str(self._coef))
+            if self._coef < 0:
+                ostream.write(")")
```

Comment:
OK.  This is a subtle but significant change (it is breaking some core tests of string generation).  We should think about this a little more...  It will require developing some unittests and adding them to the coopr3 and pyomo4 expression testing systems, then we should decide how to best implement this so that it is as least disruptive as possible.